corgidrp/measure_companions.py

```diff
@@ -0,0 +1,450 @@
+        guesssep (float): Initial guess at companion separation (in pixels).
+        guesspa (float): Initial guess at companion position angle (in degrees).
+        guessflux (float): Initial guess at companion flux scaling (dimension depends on how
+            off_axis_psf has been normalized).
+        outputdir (str): Directory to store intermediate KLIP-FM outputs (optional).
+        fileprefix (str): Prefix for output files.
+        numbasis (list of int): List of KL basis sizes to use.
+        method ({"mcmc", "maxl"}): “mcmc” = Bayesian (emcee) MCMC fit,
+            “maxl” = frequentist maximum-likelihood fit
+        annulus_halfwidth (float): Half-width of the annulus (in pixels) around `guesssep` in 
+            which KLIP will be performed.
+        movement (float): Angular exclusion criterion for PSF subtraction in pyKLIP.
+        stamp_size (int): Size of the extracted stamp (in pixels) around the companion for 
+            fitting.
+        corr_len_guess (float): Initial guess of the Gaussian Process correlation length.
+        x_range (float): Bounds for x/y offsets in linear space (uniform prior).
+        y_range (float): Bounds for x/y offsets in linear space (uniform prior).
+        flux_range (float): Bounds for log10 priors on flux length.
+        corr_len_range (float): Bounds for log10 priors on correlation length.
+        nwalkers (int): MCMC hyperparameters if method="mcmc".
+        nburn (int): MCMC hyperparameters if method="mcmc".
+        nsteps (int): MCMC hyperparameters if method="mcmc".
+        numthreads (int): Parallel threads for MCMC.
+        star_center_err (float): Uncertainty on star center (in pixels), for post-fit error 
+            propagation.
+        platescale (float): Plate scale in mas/pixel, for converting pixel offsets to mas.
+        platescale_err (float): Plate scale uncertainty in mas/pixel.
+        pa_offset (float): Known absolute offset in instrument position angle (in degrees).
+        pa_uncertainty (float): Additional systematic uncertainty in position angle (in degrees).
+
+    Returns:
+        fit (pyklip.fitpsf.FMAstrometry): The fitting object containing best-fit parameters, 
+            chains (if MCMC),and raw/final astrometry with uncertainties.
+    """
+    if not os.path.exists(outputdir):
+        os.makedirs(outputdir)
+
+    # 1) Wrap coronagraphic_dataset in PyKLIP dataset
+    coronagraphic_dataset = PyKLIPDataset(coronagraphic_dataset, highpass=False)
+
+    # 2) Prepare the 'psfs' argument for FMPlanetPSF
+    #    off_axis_psf might be 2D or 3D (if multi-wavelength)?
+    psf_array = off_axis_psf.data 
+    dataset_shape = coronagraphic_dataset.input.shape 
+    dataset_shape = coronagraphic_dataset.input.shape[-2:]
+
+    # May (or may not) need multiple wavelength channels:
+    # For now assume single-channel data:
+    # Otherwise, replace wvs=[some_array_of_wavelengths].
+    wvs = [1.0]  # placeholder single wavelength
```

Comment:
I don't think we'll need multiple wavelengths, so can always fake to 1 wavelength (unlikely we'll use the same function for spectroscopy).